### PR TITLE
feat(apig): support a new throttling policy associate resource

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add a new parameter (publish_id) to the throttling policy resource.
2. support a new policy associate resource.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccThrottlingPolicyAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccThrottlingPolicyAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccThrottlingPolicyAssociate_basic
=== PAUSE TestAccThrottlingPolicyAssociate_basic
=== CONT  TestAccThrottlingPolicyAssociate_basic
--- PASS: TestAccThrottlingPolicyAssociate_basic (467.47s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      467.590s
```

### CODE REVIEW

File: huaweicloud/services/apig/resource_huaweicloud_apig_api_publishment.go [hunk 1]
```diff
@@ -190,17 +195,17 @@ func GetVersionHistories(c *golangsdk.ServiceClient, instanceId, envId, apiId st
 	return histories, nil
 }
 
-func getCertainPublishInfo(resp []apis.ApiVersionInfo) (*apis.ApiVersionInfo, diag.Diagnostics) {
+func getCertainPublishInfo(resp []apis.ApiVersionInfo) (*apis.ApiVersionInfo, error) {
 	if len(resp) == 0 {
-		return nil, fmtp.DiagErrorf("The API does not have any published information.")
+		return nil, fmtp.Errorf("The API does not have any published information.")
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/apig/resource_huaweicloud_apig_api_publishment.go:200]: there are fmt and fmtp in this file, don't use them together

File: docs/resources/apig_throttling_policy_associate.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+* `policy_id` - (Required, String, ForceNew) Specifies the ID of the API group to which the API response belongs to.
+  Changing this will create a new resource.
+
+* `publish_ids` - (Required, List) Specifies the publish ID corresponding to the API bound by the throttling policy.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - Resource ID.
```
Comment (Jason-Zhang9309) [inline docs/resources/apig_throttling_policy_associate.md]: add a description of the ID format

File: huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy_associate.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+	d.SetId(fmt.Sprintf("%s/%s", d.Get("instance_id").(string), d.Get("policy_id").(string)))
+
+	return resourceThrottlingPolicyAssociateRead(ctx, d, meta)
+}
+
+func buildListOpts(instanceId, policyId string) throttles.ListBindOpts {
+	return throttles.ListBindOpts{
+		InstanceId: instanceId,
+		ThrottleId: policyId,
+		Limit:      500,
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy_associate.go:89]: can we make sure this can get all publish_ids?

Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy_associate.go:89]: five hundred is much big under a instance and the throttling policy.

File: huaweicloud/services/apig/resource_huaweicloud_apig_api_publishment.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_api_publishment.go:200]: This code will be fixed when it is submitted for review in the future.

File: docs/resources/apig_throttling_policy_associate.md (hunk 2 above)
Comment (Lance52259) [reply docs/resources/apig_throttling_policy_associate.md:47]: just a hash code, no format.

Comment (Lance52259) [reply docs/resources/apig_throttling_policy_associate.md]: done.

File: huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy_associate.go [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+func resourceThrottlingPolicyAssociateImportState(_ context.Context, d *schema.ResourceData,
+	_ interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.SplitN(d.Id(), "/", 2)
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid format specified for import id, must be <instance_id>/<policy_id>")
+	}
+
+	instanceId := parts[0]
+	policyId := parts[1]
+	d.SetId(fmt.Sprintf("%s/%s", instanceId, policyId))
```
Comment (hmbig2) [inline huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy_associate.go]: unnecessary,  pls remove SetId

Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy_associate.go]: sure, this format and imported are same.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccThrottlingPolicyAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccThrottlingPolicyAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccThrottlingPolicyAssociate_basic
=== PAUSE TestAccThrottlingPolicyAssociate_basic
=== CONT  TestAccThrottlingPolicyAssociate_basic
--- PASS: TestAccThrottlingPolicyAssociate_basic (509.54s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      509.704s
```

File: huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy_associate.go [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+		PublishIds: publishIds,
+	}
+}
+
+func resourceThrottlingPolicyAssociateCreate(ctx context.Context, d *schema.ResourceData,
+	meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.ApigV2Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating HuaweiCloud APIG v2 client: %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy_associate.go]: without provider(HuaweiCloud) name

File: docs/resources/apig_throttling_policy_associate.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region where the API instance and throttling policy are located.
+  If omitted, the provider-level region will be used. Changing this will create a new resource.
+
+* `instance_id` - (Required, String, ForceNew) Specifies the ID of the APIG dedicated instance to which the APIs and the
+throttling policy belongs. Changing this will create a new resource.
```
Comment (ShiChangkuo) [inline docs/resources/apig_throttling_policy_associate.md]: please add two spaces

File: huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy_associate.go [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+	}
+
+	opt := buildPolicyAssociateOpts(d.Get("instance_id").(string), d.Get("policy_id").(string),
+		utils.ExpandToStringListBySet(d.Get("publish_ids").(*schema.Set)))
+	_, err = throttles.Bind(client, opt)
+	if err != nil {
+		return diag.Errorf("error binding policy to the API: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", d.Get("instance_id").(string), d.Get("policy_id").(string)))
```
Comment (ShiChangkuo) [inline huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy_associate.go]: suggest to use two variables to save instanceID and policyID

File: huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy_associate.go (hunk 5 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy_associate.go]: done.

File: docs/resources/apig_throttling_policy_associate.md (hunk 6 above)
Comment (Lance52259) [reply docs/resources/apig_throttling_policy_associate.md]: done.

File: huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy_associate.go (hunk 7 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_throttling_policy_associate.go]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccThrottlingPolicyAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccThrottlingPolicyAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccThrottlingPolicyAssociate_basic
=== PAUSE TestAccThrottlingPolicyAssociate_basic
=== CONT  TestAccThrottlingPolicyAssociate_basic
--- PASS: TestAccThrottlingPolicyAssociate_basic (507.61s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      507.686s
```

**Pull request merged**